### PR TITLE
Add cache directories  for download jars and sonar plugin of Maven in  travis

### PR DESCRIPTION
**Pull request by ycheng9**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -103,6 +103,12 @@ matrix:
       dist: trusty
       jdk: oraclejdk11
 
+cache:
+  directories:
+    - '$HOME/.m2/repository'
+    - '$HOME/.sonar/cache'
```
Comment (jixuan1989) [inline .travis.yml]: why sonar？

**Comment by jixuan1989** [comment]
I consider that m2 and sonar caches are used for speeding up Maven test.

So, can we also speed up the download of JDK?

**Comment by MyXOF** [comment]
Normally, it takes about 5 minutes to execute `mvn clean integration-test` command. If it takes more than 5 minutes, downloading jdk becomes a bottleneck of our travis test speed.

image

**Comment by jixuan1989** [comment]
I will merge this PR. 
Cache downloading JDK can be submitted in a new PR.

**Pull request merged**